app/models/spree/invoice.rb

```diff
@@ -0,0 +1,128 @@
+class Spree::Invoice < ActiveRecord::Base
+  belongs_to :invoiceable, polymorphic: true
+  validates :invoiceable, presence: true
+
+  before_save :set_date
+  before_save :set_number, if: :use_sequential_number?
+  after_save :increase_invoice_number, if: :use_sequential_number?
+
+  # Delegates
+
+  delegate :bill_address,
+           :ship_address,
+           :tax_address,
+           :line_items,
+           :display_item_total,
+           :all_adjustments,
+           :shipments,
+           :display_total,
+           :payments, to: :invoiceable
+
+  # Delegate #number to the invoiceable if we don't generate a number ourselves
+  #
+  def number
+    if use_sequential_number?
+      super
```

Comment:
No, it's supposed to read the number from the DB. I realize I even wrote a test for that, but the test is meaningless. Thanks, no, that is not intended.